lib/truffle/truffle/patching.rb

```diff
@@ -1,140 +1,69 @@
-require 'pathname'
-
 module Truffle::Patching
-  DIR     = Pathname(Truffle::Boot.ruby_home).join("lib/patches")
-  # Allowed operations are: :before, :after, :instead, :ignore
-  PATCHES = { "bundler"                                 => :before,
-              "bundler/version"                         => :after,
-              "bundler/cli/exec"                        => :after,
-              "bundler/compact_index_client/updater"    => :after,
-              "bundler/current_ruby"                    => :after,
-              "bundler/dependency"                      => :after,
-              "bundler/fetcher/downloader"              => :after,
-              "bundler/fetcher/compact_index"           => :after,
-              "bundler/ruby_version"                    => :after,
-              "bundler/source/rubygems"                 => :after,
-
-              "rubygems"                                => :before,
-              "rubygems/ext"                            => :after,
-              "rubygems/package"                        => :after,
-              "rubygems/remote_fetcher"                 => :after,
-              "rubygems/request"                        => :after,
-              "rubygems/request_set/gem_dependency_api" => :after,
-
-              "rspec/support"                           => :after,
-
-              "unf/normalizer_cruby"                    => :before,
-  }
-
-  PATCHES.each do |file, _|
-    unless DIR.join(file).sub_ext('.rb').exist?
-      raise "file #{file} is missing in #{DIR}"
+  extend self
+
+  def patches
+    @patches ||= begin
+      require 'pathname'
+      Pathname.glob(dir.join('*')).reduce({}) do |patches, path|
+        next patches unless path.directory?
+        patches.update path.basename.to_s => path
+      end
```

Comment:
I personally don't like this style of pseudo functional code.
If it's side-effecting, I prefer to just use each, it's much simpler:
```ruby
require 'pathname'
patches = {}
Pathname.glob(dir.join('*')) do |path|
  patches[path.basename.to_s] = path if path.directory?
end
patches
```
Also pathname is overkill here (and the resulting `path` is only used as a String in `insert_patching_dir`):
```ruby
patches = {}
Dir.glob("#{dir}/*") do |path|
  patches[File.basename(path)] = path if File.directory?(path)
end
patches
```